### ISSUE TITLE
explicit-module-boundary-types and JavaScript files

### ISSUE THREAD

**Issue by Conaclos** [denoland/deno_lint#1049]
Hi!

### Lint Name

[explicit-module-boundary-types](https://lint.deno.land/?q=&all=on#explicit-module-boundary-types)

### Code Snippet

```js
export function f() {}
```

```json
{
  "lint": {
    "files": {
      "include": ["src/"]
    },
    "rules": {
      "tags": ["recommended"],
      "include": ["explicit-module-boundary-types"]
    }
  }
}
```

### Expected Result

The rule should not be applied on JavaScript files.

### Actual Result

```
export function f() {}
       ^^^^^^^^^^^^^^^
    src/index.js:1:7

    hint: Add a return type to the function signature
    help: for further information visit https://lint.deno.land/#explicit-function-return-type
```

### Version

deno 1.21.3 (release, x86_64-unknown-linux-gnu)
v8 10.0.139.17
typescript 4.6.2

I also tested with @node-rs/denolint 1.13.1 (release, x86_64-unknown-linux-gnu)
(The previous diagnostic comes from there)
